src/xcode/ENA/ENA/Source/Views/BaseElements/ENAButton.swift

```diff
@@ -15,55 +15,148 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import Foundation
 import UIKit
 
-/// A Button which has the same behavior of UIButton, but with different tint color.
 @IBDesignable
-class ENAButton: UIButton {
-	override init(frame: CGRect) {
-		super.init(frame: frame)
-		customizeButton()
+class ENAButton: DynamicTypeButton {
+	@IBInspectable var color: UIColor?
+
+	@IBInspectable var isTransparent: Bool = false { didSet { applyStyle() } }
+	@IBInspectable var isLight: Bool = false { didSet { applyStyle() } }
+	@IBInspectable var isInverted: Bool = false { didSet { applyStyle() } }
+
+	private var style: Style = .regular
+
+	override var isEnabled: Bool { didSet { applyStyle() } }
+	override var isHighlighted: Bool { didSet { applyHighlight() } }
+
+	private var highlightView: UIView!
+
+	override var intrinsicContentSize: CGSize {
+		var size = super.intrinsicContentSize
+		if size.height < 50 { size.height = 50 }
+		return size
 	}
 
 	required init?(coder: NSCoder) {
 		super.init(coder: coder)
-		customizeButton()
+		setup()
 	}
 
-	override var isEnabled: Bool {
-		didSet {
-			if isEnabled {
-				backgroundColor = UIColor.preferredColor(for: .tint)
-			} else {
-				backgroundColor = UIColor(red: 0.96, green: 0.96, blue: 0.96, alpha: 1.00)
-			}
-		}
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+		setup()
 	}
 
-	private func customizeButton() {
-		setButtonColors()
-		setRoundCorner(radius: 8.0)
+
+	override func prepareForInterfaceBuilder() {
+		setup()
+		super.prepareForInterfaceBuilder()
+	}
+
+
+	override func awakeFromNib() {
+		setup()
+		super.awakeFromNib()
+	}
+
+	private func setup() {
+		setValue(ButtonType.custom.rawValue, forKey: "buttonType")
+
+		clipsToBounds = true
 
 		contentEdgeInsets = UIEdgeInsets(top: 14, left: 14, bottom: 14, right: 14)
 		heightAnchor.constraint(greaterThanOrEqualToConstant: 50).isActive = true
+		titleLabel?.numberOfLines = 0 // TODO
+
+		highlightView?.removeFromSuperview()
+		highlightView = UIView(frame: bounds)
+		highlightView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+		addSubview(highlightView)
 
-		titleLabel?.adjustsFontForContentSizeCategory = true
+		cornerRadius = 8
```

Comment:
I would prefer standard corner radius of six points
https://developer.apple.com/design/human-interface-guidelines/watchos/interface-elements/buttons/

Reply:
Actually this should have been 10 according to the design specs. The page you linked is for watchOS only. iOS itself does not know buttons as we use them here at all.